setup.py

```diff
@@ -258,6 +258,19 @@ def build_js_files(self):
     undef_macros=UNDEF_MACROS,
 )
 
+MEMRAY_INJECT_EXTENSION = Extension(
+    name="memray._inject",
+    sources=[
+        "src/memray/_memray/inject.cpp",
+    ],
+    language="c++",
```

Comment:
More bad news on MacOS, if debug info is not available, lldb refuses to do anything: if I remove the `.o` that contain the DWARF then `attach` fails.

Reply:
We need to run `dsymutil` on the resulting shared object **while** the `.o` are still available and make sure that's packed with the extension

Reply:
Interestingly debugpy doesn't have this problem. This is how they do it:

https://github.com/microsoft/debugpy/blob/bc9c596599bafc1ef0629124a2b0196e2823f6f3/src/debugpy/_vendored/pydevd/pydevd_attach_to_process/linux_and_mac/lldb_prepare.py#L36

Reply:
Ah, it seems that `lldb` is ok if `spawn_client` is declared outside the `memray` namespace as a `extern "C"` symbol.

Reply:
The fix we're going with is making the function `extern "C"` and outside the `memray` C++ namespace, and casting it to the right type before calling it so that we don't depend on DWARF for knowing the signature.